### ISSUE TITLE
AndroidGraphicsLiveWallpaper ANR on resume 

### ISSUE THREAD

**Issue by oshetima** [libgdx/libgdx#5276]
#### Issue details
Hello,
I want to show here an ANR that I'm getting from several times on my live wallpaper apps. It's a deadlock on the resume method of AndroidGraphicsLiveWallpaper class. It is impossible for me to reproduce and it happens randomly on my users.

#### Version of LibGDX and/or relevant dependencies
Version: 1.9.8.   

#### Stacktrace
```java

"main" tid=1 Waiting
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x75796250 self=0xa6784400
  | sysTid=4483 nice=0 cgrp=default sched=0/0 handle=0xa9428534
  | state=S schedstat=( 30824104337 11300023843 31183 ) utm=2744 stm=338 core=3 HZ=100
  | stack=0xbe3c1000-0xbe3c3000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0c8f2468> (a java.lang.Object)
  at com.badlogic.gdx.backends.android.AndroidGraphicsLiveWallpaper.resume (AndroidGraphicsLiveWallpaper.java:139)
- locked <0x0c8f2468> (a java.lang.Object)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaper.onResume (AndroidLiveWallpaper.java:159)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onResume (AndroidLiveWallpaperService.java:402)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.notifyVisibilityChanged (AndroidLiveWallpaperService.java:369)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onVisibilityChanged (AndroidLiveWallpaperService.java:361)
  at android.service.wallpaper.WallpaperService$Engine.reportVisibility (WallpaperService.java:925)
  at android.service.wallpaper.WallpaperService$Engine$3.onDisplayChanged (WallpaperService.java:1061)
  at android.hardware.display.DisplayManagerGlobal$DisplayListenerDelegate.handleMessage (DisplayManagerGlobal.java:462)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:154)
  at android.app.ActivityThread.main (ActivityThread.java:6119)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:886)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:776)
```

#### Please select the affected platforms
- [x] Android

It looks like the same problem has been reported on issue #4267 and  #5042 but I see no solution implemented till now. If someone has found a workaround to this please let me know because it is really frustrating and I'm loosing visibility on my apps day after day thanks to this ANR (It is present on all my live wallpaper apps). Any help is appreciated, many thanks in advance.

**Comment by Tom-Ski** [comment]
Please do not file duplicate issues.

**Comment by urnish** [comment]
Hi @ArdiDev,

Have you found any solution regarding the above issues, which you mentioned, because same issues I am getting in the Android app which I am developing.

Answers are welcome

**Comment by tommyettinger** [comment]
@urnish This issue is closed. It has been closed for nearly **7 years**. I find it hard to believe ArdiDev is still developing the same app today! If you had done just a little more reading, you would have noticed Tom-Ski's comment above saying this is a duplicate issue, and looked into the linked issues, which I actually did, just now! I found https://github.com/libgdx/libgdx/pull/6074 , which was merged almost **5 years ago** and even then people were already saying the reports had mostly stopped coming in. So, the *appropriate* thing to do would be to either comment on the merged PR saying it doesn't appear to work as intended and *giving your full information in the same manner as you would on a new issue*, or better yet, just make a new issue and follow the issue template. We don't know what version of libGDX you're using, but it sounds like it might be a very old one. If you can replicate with version 1.13.1, make a new issue and provide all information you can, because it almost certainly wouldn't be the same (fixed!) 7-year-old issue. Provide any stacktrace you can, too, because the above issue won't have meaningful line numbers this much time later.

**Comment by urnish** [comment]
@tommyettinger, thanks a lot for so quick response, and I really appreciate the details you provided. Actually, I also reviewed the different PR and couldn't find any solid solution, so I thought to post the issues.

I am currently using 1.13.0 and getting these issues. Apart from that, I have mentioned the dependency that I am currently using.

Let me know If I am wrong and making any mistakes.


 implementation("com.badlogicgames.gdx:gdx-backend-android:1.13.0")
    implementation("com.badlogicgames.gdx:gdx:1.13.0")

![Image](https://github.com/user-attachments/assets/5bf1a95f-44e6-466f-915f-f901e5faedb7)

**Comment by tommyettinger** [comment]
Well something is definitely wrong there... Version 1.13.0 didn't have any armeabi natives, so you have a mix of versions in there, and one is definitely older. See https://repo1.maven.org/maven2/com/badlogicgames/gdx/gdx-platform/1.13.0/ for the list of natives included in 1.13.0 . Also, 1.13.0 was a rather buggy release, and 1.13.1 came out not much later because the fixes were ready quickly and were needed quickly. First thing to do is to clean your project as deeply as you can, to get rid of native libraries that are probably conflicting in nasty ways (and aren't even used by Google Play Store anymore). Next thing is probably to update libGDX to 1.13.1; check [the 'CHANGES' file](https://github.com/libgdx/libgdx/blob/master/CHANGES#L25-L55) for the list of what changed (there's only one breaking change affecting Android, and it's mostly easier to use Android with 1.13.1 than 1.13.0). You'll probably want to update your Android Gradle Plugin version and see what min/target SDK you want to use at this point in time, since those are always changing. If you have problems with the project structure not cooperating, you can try making a new empty project with [gdx-liftoff](https://github.com/libgdx/gdx-liftoff/releases), selecting at least the Android platform and ideally LWJGL3 too, then copying in all of your old Java or Kotlin or whatever code into that skeletal project to see if it works there. You might need to use the new skeletal build.gradle files as a model for what to bring over from your older ones, like any dependencies beyond libGDX will probably just go in core/build.gradle , and some Android-specific deps may go in android/build.gradle , but in both cases *not* in the "buildscript" section (that only contains deps for Gradle to use at build-time, not for your app itself). The nice thing about Liftoff creating a new Android project is that you don't have to use Android Studio's cumbersome "update assistant" to change each version to the current recommendations.

**Comment by urnish** [comment]
@tommyettinger, thanks again for your quick response. I have upgraded to 1.13.1  and apart from that, deleted the "libs" folder and added the dependency as shown in the screenshot. 

But after adding this one, I am getting crash which also mentioned in the screenshot, which never actually faced is use ".so" file from "libs" folder. 

Any idea about this one?

![Image](https://github.com/user-attachments/assets/d674fa48-8757-46e0-86f1-47464d50947d)
![Image](https://github.com/user-attachments/assets/ff45d90e-ddc7-4d98-b50a-7004ad3c5030)

**Comment by tommyettinger** [comment]
How about we move the conversation to #7620 , which I just saw you had opened 2 hours ago... I'm sure Tom-ski and ArdiDev don't appreciate the email barrage the help topic here is sending them! 😄

**Comment by urnish** [comment]
That's perfectly fine @tommyettinger. apprecite the suggestion.

**Comment by urnish** [comment]
@tommyettinger sorry for disturbing again but want to know some suggestion as mentioned in #7620
